src/libs/SelectionScraper/base.js

```diff
@@ -0,0 +1,161 @@
+                    } else {
+                        node = range.commonAncestorContainer.parentNode.closest(`[${tagAttribute}]`);
+                    }
+
+                    // This means "range.commonAncestorContainer" is a text node. We simply get its parent node.
+                    if (!node) {
+                        node = range.commonAncestorContainer.parentNode;
+                    }
+
+                    node = node.cloneNode();
+                    node.appendChild(clonedSelection);
+                    div.appendChild(node);
+                }
+            }
+
+            return div.innerHTML;
+        }
+
+        return window.getSelection().toString();
+    }
+
+    // If browser doesn't support Selection API, returns empty string.
+    return '';
+};
+
+/**
+ * Clears all attributes from dom elements
+ * @param {Object} dom htmlparser2 dom representation
+ * @returns {Object} htmlparser2 dom representation
+ */
+const replaceNodes = (dom) => {
+    let domName = dom.name;
+    let domChildren;
+    const domAttribs = {};
+    let data;
+
+    // Encoding HTML chars '< >' in the text, because any HTML will be removed in stripHTML method.
+    if (dom.type === 'text') {
+        data = Str.htmlEncode(dom.data);
+    }
+
+    // We are skipping elements which has html and body in data-testid, since ExpensiMark can't parse it. Also this data
+    // has no meaning for us.
+    if (dom.attribs && dom.attribs[tagAttribute]) {
+        if (!elementsWillBeSkipped.includes(dom.attribs[tagAttribute])) {
+            domName = dom.attribs[tagAttribute];
+        }
+
+        // Adding a new line after each comment here, because adding after each range is not working for chrome.
+        if (dom.attribs[tagAttribute] === 'comment') {
+            dom.children.push(new Element('br', {}));
+        }
+    }
+
+    // We need to preserve href attribute in order to copy links.
+    if (dom.attribs && dom.attribs.href) {
+        domAttribs.href = dom.attribs.href;
+    }
+
+    if (dom.children) {
+        domChildren = _.map(dom.children, c => replaceNodes(c));
+    }
+
+    return {
+        ...dom,
+        data,
+        name: domName,
+        attribs: domAttribs,
+        children: domChildren,
+    };
+};
+
+/**
+ * Reads the provided html and converts it to the return types.
+ * @param {Object} selectedHtml html to parse
+ * @returns {Object} converted text, html, and markdown
+ */
+const getCustomAsTypes = (selectedHtml) => {
+    if (!selectedHtml) {
+        return null;
+    }
+    const domRepresentation = parseDocument(selectedHtml);
+    domRepresentation.children = _.map(domRepresentation.children, c => replaceNodes(c));
+
+    const newHtml = render(domRepresentation);
+
+    const parser = new ExpensiMark();
+
+    const text = parser.htmlToText(newHtml);
+    const markdown = parser.htmlToMarkdown(newHtml);
+    return {html: `<html><body>${newHtml}</body></html>`, text, markdown};
```

Comment:
Let's just take out the markdown from it. Rename this function to `getSelection` which returns an object of `{html, text}`. Handle HTML to MD conversion where this is needed.